sagemathinc/cocalc
# desktop notifications POC

# Description
I have started using webpages in full screen because I have a small laptop. I would like to know wheb people chat at me, even if I'm not currently looking at CoCalc (which happens sometimes). Desktop notifications seem like they would solve this.

# Testing Steps
1. Open a project with account A
1. Open the same project with a different account B
1. Account B does something to increase the notification count for account A (work on a shared file)
1. Account A should get a desktop notification

# Relevant Issues
Some people find desktop notifications extremely intrusive. This code aims to be polite, but I'm pretty sure it could still lead to a blow-up with lots of notifications. There is a lot of ways you could lessen the amount of notifications: show up to 1 per minute, show one per project, show one per person, etc. I can probably edit the PR if the official CoCalc team (a) actually wants or cares abut this feature, and (b) has a preferred plan to limit notifications.

### [Checklist](https://github.com/sagemathinc/cocalc/wiki/PR-Checklist):

Front end:
